src/accelerate/data_loader.py

```diff
@@ -365,6 +365,9 @@ def prepare_data_loader(
 
     new_dataset = dataloader.dataset
     new_batch_sampler = dataloader.batch_sampler
+    if isinstance(new_dataset, IterableDataset):
+        # Iterable dataset doesn't like batch_sampler, data_loader creates a default one for it
+        new_batch_sampler = None
```

Comment:
```suggestion
    # Iterable dataset doesn't like batch_sampler, but data_loader creates a default one for it
    new_batch_sampler = dataloader.batch_sampler if not isinstance(new_dataset, IterableDataset) else None
```
Let's group the two lines together.